core/src/main/java/tachyon/client/TachyonFile.java

```diff
@@ -204,10 +204,13 @@ public String getLocalFilename(int blockIndex) throws IOException {
    * @throws IOException
    */
   public List<String> getLocationHosts() throws IOException {
+    List<String> ret = new ArrayList<String>();
+    if (getNumberOfBlocks() == 0) {
+      return ret;
+    }
```

Comment:
Instead of having an early return here, could we wrap the next block in an if (getNumberOfBlocks() != 0)? Requires more indentation, but I think it makes it easier to read?